# Task2 : Data cleaning and preprocessing
### By: Priyanka Choudhary

In [10]:
import pandas as pd
import numpy as np

In [11]:
df = pd.read_csv('../data/dataset.csv')
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [12]:
print("Missing values before cleaning")
df.isnull().sum()

Missing values before cleaning


PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [13]:
df['Age'].fillna(df['Age'].median(), inplace=True)
print("Age missing values after fix:",df['Age'].isnull().sum())

Age missing values after fix: 0


In [14]:
df['Embarked'].fillna(df['Embarked'].mode()[0], inplace = True)
print("Embarked missing values after fix:", df['Embarked'].isnull().sum())

Embarked missing values after fix: 0


In [15]:
df.drop(columns= ['Cabin'],inplace = True )
print("Cabin column dropped")

Cabin column dropped


In [16]:
print("Duplicated before cleaning:", df.duplicated().sum())

Duplicated before cleaning: 0


In [17]:
print("Dataset shape after cleaning:", df.shape)

Dataset shape after cleaning: (891, 11)


## Summary
- Filled 177 missing Age values with median (28)
- Filled 2 missing Embarked values with mode (S)
- Dropped Cabin column (77% data missing)
- No duplicate rows found
- Final dataset shape: 891 rows × 11 columns